pandas/tests/arrays/floating/test_to_numpy.py

```diff
@@ -0,0 +1,123 @@
+import numpy as np
+import pytest
+
+import pandas as pd
+import pandas._testing as tm
+
+
+@pytest.mark.parametrize("box", [True, False], ids=["series", "array"])
+def test_to_numpy(box):
+    con = pd.Series if box else pd.array
+
+    # default (with or without missing values) -> object dtype
+    arr = con([0.1, 0.2, 0.3], dtype="Float64")
+    result = arr.to_numpy()
+    expected = np.array([0.1, 0.2, 0.3], dtype="object")
+    tm.assert_numpy_array_equal(result, expected)
+
+    arr = con([0.1, 0.2, None], dtype="Float64")
+    result = arr.to_numpy()
+    expected = np.array([0.1, 0.2, pd.NA], dtype="object")
+    tm.assert_numpy_array_equal(result, expected)
+
+
+@pytest.mark.parametrize("box", [True, False], ids=["series", "array"])
+def test_to_numpy_float(box):
+    con = pd.Series if box else pd.array
+
+    # no missing values -> can convert to float, otherwise raises
+    arr = con([0.1, 0.2, 0.3], dtype="Float64")
+    result = arr.to_numpy(dtype="float64")
+    expected = np.array([0.1, 0.2, 0.3], dtype="float64")
+    tm.assert_numpy_array_equal(result, expected)
+
+    arr = con([0.1, 0.2, None], dtype="Float64")
+    with pytest.raises(ValueError, match="cannot convert to 'float64'-dtype"):
+        result = arr.to_numpy(dtype="float64")
+
+    # need to explicitly specify na_value
+    result = arr.to_numpy(dtype="float64", na_value=np.nan)
+    expected = np.array([0.1, 0.2, np.nan], dtype="float64")
+    tm.assert_numpy_array_equal(result, expected)
+
+
+@pytest.mark.parametrize("box", [True, False], ids=["series", "array"])
+def test_to_numpy_int(box):
+    con = pd.Series if box else pd.array
+
+    # no missing values -> can convert to int, otherwise raises
+    arr = con([1.0, 2.0, 3.0], dtype="Float64")
+    result = arr.to_numpy(dtype="int64")
+    expected = np.array([1, 2, 3], dtype="int64")
+    tm.assert_numpy_array_equal(result, expected)
+
+    arr = con([1.0, 2.0, None], dtype="Float64")
+    with pytest.raises(ValueError, match="cannot convert to 'int64'-dtype"):
+        result = arr.to_numpy(dtype="int64")
+
+    # automatic casting (floors the values)
+    arr = con([0.1, 0.9, 1.1], dtype="Float64")
+    result = arr.to_numpy(dtype="int64")
+    expected = np.array([0, 0, 1], dtype="int64")
+    tm.assert_numpy_array_equal(result, expected)
+
+
+@pytest.mark.parametrize("box", [True, False], ids=["series", "array"])
+def test_to_numpy_na_value(box):
+    con = pd.Series if box else pd.array
+
+    arr = con([0.0, 1.0, None], dtype="Float64")
+    result = arr.to_numpy(dtype=object, na_value=None)
```

Comment:
Do you need to add tests here (now or maybe later pending the `np.nan`/`NA` decision) with `na_value=np.nan` ?

Reply:
There is already one covering that above in `test_to_numpy_float` (however, what would probably still be useful to add is a case where we have both NaN and NA in the FloatingArray, and then use an `na_value` that is *not* np.nan, to ensure we only convert the NA)

Reply:
Added a test